keps/sig-node/3815-pod-condition-for-configuration-errors/README.md

```diff
@@ -0,0 +1,1069 @@
+
+## Motivation
+
+Batch users may not have deep knowledge of Kubernetes.  It is common occurrence to find users submitting Pods with configuration errors (invalid images, wrong image names, missing volumes, missing secrets, wrong volume/secret for config map).  These cases cause the Pod to be stuck in pending and there is code required to remove these pods or user input to remove.
+
+As a batch system developer, we find that higher level controllers have to write code to handle Pending Pod cases.  Since the pods are not considered failed, the higher level controller needs to transition this state to failed in order to enforce retries.  For example, the job controller will run these pods and the state of the Pod will be in Pending.  The job controller does not transition this state to failed even if the pod will never start running.
+
+For CNCF projects, in the Armada project, we implement logic that determines retry ability based on events and statuses.  Events are not standard so we expect issues when updating and we discovered various states that don't have a representative status.  
+
+It would be ideal to have a standard condition for these common configuration errors.  These errors can be fixed via human interaction but they can sometimes be difficult to programatically fix.  A goal of having a standard condition for these configuration errors would be for third-party controllers and/or Kubernetes controllers to start using a single way of detecting fixable configuration errors.
+
+<!--
+This section is for explicitly listing the motivation, goals, and non-goals of
+this KEP.  Describe why the change is important and the benefits to users. The
+motivation section can optionally provide links to [experience reports] to
+demonstrate the interest in a KEP within the wider Kubernetes community.
+
+[experience reports]: https://github.com/golang/go/wiki/ExperienceReports
+-->
+
+### Goals
+
+- Add a new condition FailingToStart to detect failures in configuration.
+
+For future cases that we want to add to to this condition, the suggestion would be to consider cases where pods are forever stuck in the pending state.  We want to avoid transient pods that will eventually resolve themselves so we decided to avoid targeting events that are related to ImagePullBackOff because these can eventually be fixed without any interaction.  The general rule could be that pods that require manually intervention to fix could be added as cases for this condition.  
```

Comment:
ImagePullBackoff *could* require intervention (we don't know without other information which is buried layers away).  IMO, we shouldn't shy away from those cases - they are just as "failing to start" as the others.

Reply:
And to be clear, we don't need to handle 100% of errors right away, but we will want to know what this is for, so we can decide to add or not-add others over time (though we should get as complete as we can afford as quickly as possible, obviously :)

Reply:
In some cases, the user uses other components(in most cases, they use pre-pull, which would be OK.) to pull images(or load images from storage), and the imagePullPolicy is `ifNotPresent`. The pod can be started when the image is ready. For them, this would be a behavior change.

Reply:
Circling back to this, one thing I notice about `ImagePullBackOff` is that it seems to be a generic error.  For example, if you have an invalid `imagePullSecret` you will get a `ImagePullErr` for a short while and then it will going into an image crash loop and give you `ImagePullBackOff`.  You can see the events for `ImagePullErr` but the container status changes after a while.  

So I wonder if bubbling up the condition `ImagePullErr` is useful in this case as it would be nice to not have to read events to programatically get this failure.  

I updated this KEP with the `ImagePullErr` case also.